src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -7508,48 +7477,23 @@ namespace WaterThermalTanks {
-			//HPWaterInletNodeTempSaved = Node( HPWaterInletNode ).Temp;
-			// This for loop is intended to iterate and converge on a condenser operating temperature so that the evaporator model correctly calculates performance.
-			// CHECK- embedded both the nonVS and the VS sim calls inside the for-loop, previously the VS wasn't in a for loop
-			for ( loopIter = 1; loopIter <= 4; ++loopIter ) {
-				if (MaxSpeedNum > 0) { // lowest speed of VS HPWH coil
-					SpeedRatio = 1.0;
-					HPPartLoadRatio = 1.0;
-					bIterSpeed = true; // prepare for iterating between speed levels
-					SpeedNum = 1;
-					SetVSHPWHFlowRates(WaterThermalTankNum, Tank.HeatPumpNum, SpeedNum, SpeedRatio,
-						RhoWater, MdotWater, FirstHVACIteration);
-					SimVariableSpeedCoils(HeatPump.DXCoilName, HeatPump.DXCoilNum,
-						CycFanCycCoil, EMP1, EMP2, EMP3, 1, HPPartLoadRatio, SpeedNum, SpeedRatio, 0.0, 0.0, 1.0);
-				} else {
-					CalcHPWHDXCoil(HeatPump.DXCoilNum, HPPartLoadRatio);
-				}
-				// Currently, HPWH heating rate is only a function of inlet evap conditions and air flow rate
-				// If HPWH is ever allowed to vary fan speed, this next sub should be called.
-				// CALL CalcDOE2DXCoil(DXCoilNum, HPPartLoadRatio, FirstHVACIteration,PartLoadRatio, FanOpMode)
-				// (possibly with an iteration loop to converge on a solution)
-				CondenserDeltaT = Node( HPWaterOutletNode ).Temp - Node( HPWaterInletNode ).Temp;
-				Tank.SourceInletTemp = Node(HPWaterInletNode).Temp + CondenserDeltaT;
-
-				// this CALL does not update node temps, must use WaterThermalTank variables
-				// select tank type
-				{ auto const SELECT_CASE_var1( HeatPump.TankTypeNum );
-				if ( SELECT_CASE_var1 == MixedWaterHeater ) {
-					CalcWaterThermalTankMixed( WaterThermalTankNum );
-					NewTankTemp = Tank.TankTemp;
-				} else if ( SELECT_CASE_var1 == StratifiedWaterHeater ) {
-					CalcWaterThermalTankStratified( WaterThermalTankNum, HPPartLoadRatio );
-					NewTankTemp = FindStratifiedTankSensedTemp( Tank );
-				}}
-
-				LowSpeedTankTemp = NewTankTemp;
-
-				Node( HPWaterInletNode ).Temp = Tank.SourceOutletTemp;
-				if ( std::abs( Node( HPWaterInletNode ).Temp - HPWHCondInletNodeLast ) < SmallTempDiff ) break;
-				HPWHCondInletNodeLast = Node( HPWaterInletNode ).Temp;
+			// Do the coil and tank calculations at full PLR to see if it overshoots setpoint.
+			if ( MaxSpeedNum > 0 ) { // lowest speed of VS HPWH coil
+				SpeedRatio = 1.0;
+				HPPartLoadRatio = 1.0;
+				bIterSpeed = true; // prepare for iterating between speed levels
+				SpeedNum = 1;
+				SetVSHPWHFlowRates( WaterThermalTankNum, Tank.HeatPumpNum, SpeedNum, SpeedRatio, RhoWater, MdotWater, FirstHVACIteration );
+				SimVariableSpeedCoils( HeatPump.DXCoilName, HeatPump.DXCoilNum, CycFanCycCoil, EMP1, EMP2, EMP3, 1, HPPartLoadRatio, SpeedNum, SpeedRatio, 0.0, 0.0, 1.0 );
+				CalcWaterThermalTank( WaterThermalTankNum );
+			} else {
+				ConvergeSingleSpeedHPWHCoilAndTank( WaterThermalTankNum, HPPartLoadRatio );
```

Comment:
Here I moved the loop logic into that `ConvergeSingleSpeedHPWHCoilAndTank` function because I'll need to use it elsewhere later. The variable speed stuff didn't need to be in the loop.